Iss204 -- access to health services

vpancini opened the pull request:
This metric represents the number of individuals served by one primary care physician in a county, if the population was equally distributed across physicians. Data come from the Area Health Resource File (AHRF) for 2014-2021. We then check our results for the most recent year against the results from County Health Rankings, which don't match because they used a different source of population.

vpancini commented:
Implemented a few more changes. The final file is `access-health-services-all.Rmd`. Areas for extra attention by the reviewer are marked with `todo():`.

The `access-health-services-2020.Rmd` file can be ignored.

malcalakovalski reviewed (changes requested):
Overall, this metric is very well documented and easy to reproduce. I described minor edits I had to do to render the file in my suggestions. 

One overarching suggestion I have is to add chunk labels  to make debugging easier and to consider switching from .Rmd to .qmd. I think this would be worth it since it's not a big lift to convert between both formats. 

I'm not sure what the best way to proceed on the discrepancies between counties in the Urban `county-population.csv` and AHRF so I'd suggest raising it to the research team.

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
````diff
@@ -0,0 +1,537 @@
+
+```
+
+Urban Institute and AHRF population sources match for every year except 2021.
+    todo(): probably a more elegant way to check this
+```{r}
+my_data %>%
+  mutate(pop_diff = population - pop) %>%
+  group_by(year) %>%
+  summarise(mean(pop_diff))
+
+```
````
This was super helpful! Implemented this method instead.

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
```diff
@@ -0,0 +1,537 @@
+
+
+## 5. Add data quality flags
+This section adds a flag for counties with less than 2,000 people to indicate low population, and assigns a data quality flag based on the 
```
Completed the sentence.

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
````diff
@@ -0,0 +1,537 @@
+    
+Read in Urban Institute population data.
+```{r}
+county_pop <- read.csv(here::here("04_health", "county-populations.csv")) %>%
````
Agreed. Changed.

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
````diff
@@ -0,0 +1,537 @@
+Alaska had one new county in 2021, so the 2022-2023 file (which has 2021 information for primary care physicians) has one extra observation.
+```{r}
+# Check join
+anti_join(x = ahrf22, y = ahrf23, b = c("state", "county")) %>%
````
This was just a typo. Fixed.

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
```diff
@@ -0,0 +1,537 @@
+The Area Health Resource File already includes population data from the Census County Population Estimates for each year, as well as from the 2020 Census Redistricting file for 2020, and from LAPI for 2020 and 2021 (I don't know what LAPI means). These sources are recorded in the technical documentation Excel files available after downloading the AHRF zip file as described in section 3.
+
+However, to promote consistency across all metrics created as part of the Upward Mobility Framework project, we use population data compiled by Urban Institute researchers. These data are available on our project's GitHub repository [here](https://github.com/UI-Research/mobility-from-poverty/blob/main/geographic-crosswalks/data/county-populations.csv).
+    todo(): I swear I read somewhere for this project that we should all be using this file for our population counts, but I cannot find that written down anywhere now. If true, we should be explicit about where that data is pulled from. If not true, maybe we can remove this part and just use the population data provided through AHRF.
```
Thanks for confirming!

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
````diff
@@ -0,0 +1,537 @@
+### Compare values from CHR and AHRF
+
+County Health Rankings reports a missing value for counties with population greater than 2,000 and 0 primary care physicians. We do not report a missing value for these observations. After removing observations with a missing value for PCPs from the CHR data, the number of PCPs is consistent across our data and the CHR data. This makes sense, because AHRF was the source for both.
+    todo(): The `stopifnot()` function was failing even though it is true as tested by subtracting the values and looking at the results (which were all 0)...
+```{r}
+joined_data <- joined_data %>%
+  filter(!is.na(pc_physicians_chr))
+
+# stopifnot(joined_data$pc_physicians_2020 == joined_data$pc_physicians_chr)
+
+```
````
This is unexpected, but I have addressed it.

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
````diff
@@ -0,0 +1,537 @@
+```
+
+However, the ratios are not the same, even after removing observations with 0 PCPs. This implies that County Health Rankings used a different population source than we did. We used the 2020 Decennial Census population, which was available in the AHRF file.
+    todo(): The `stopifnot()` function was failing even though it is true
+```{r}
+test <- joined_data %>%
+  filter(consequent==1)
+
+# stopifnot(test$ratio_population_pc_physician_2020 != test$ratio_population_pc_physician_chr)
+
````
I re-wrote this section to clarify what I was doing and how I was checking our ratio against CHR's

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
````diff
@@ -0,0 +1,537 @@
+
+```
+
+    todo(): There are 70 observations in the AHRF data that are not in Urban's universe of counties (from Alaska, Connecticut, and Virginia). I imagine most or all of these are county changes - how do I crosswalk these?
````
I couldn't find anything in AHRF documentation, but these are primarily counties that no longer exist as of 2014. I wonder if AHRF just includes them for consistency? Most of them had `NA` values for all variables, except one county from Alaska that had 11 deaths. They all get dropped and I have clarified why.

malcalakovalski replied:
That does make sense to me, thanks for checking!

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
```diff
@@ -0,0 +1,537 @@
+
+
+## 7. Save and write out data
+    todo(): low_pop_flag likely needs renamed but I'm not sure to what. 
```
I think I just thought that every variable relating to each metric had to follow a specific naming convention. I renamed it to adhere to the naming convention for other variables in this metric.

malcalakovalski replied:
Ah, you're right. Your renaming is better.

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
```diff
@@ -0,0 +1,537 @@
+
+For a while, we could not get our results to match the latest results from County Health Rankings. The number of primary care physicians were identical, but the ratios were not, which indicated that we used a different population source (population was not reported in CHR data). I recreated our ratio multiple times using different sources of population but could not match the CHR results.
+
+On 2024-01-31, Vincent and Lily Robin met with a team of County Health Rankings researchers. Their documentation indicates that CHR used ... One of their researchers told us that they created their ratio using Vintage 2020 population estimates, which are based on the 2010 Census (see more [here](https://www.census.gov/programs-surveys/popest/technical-documentation/research/evaluation-estimates/2 ...
```
Filled this out and clarified.

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
```diff
@@ -0,0 +1,537 @@
+                                "AHRF_SAS_2022-2023", "ahrf2023.sas7bdat"))
+
+# Remove zip file
+# unlink(zip_dest_23)
```
Uncomments + thanks for the tip!

vpancini commented on 04_health/access-health-services/access-health-services-all.Rmd:
```diff
@@ -0,0 +1,537 @@
+ahrf22 <- haven::read_sas(here::here("04_health", "access-health-services", "data", "ahrf2022.sas7bdat"))
+
+# Remove zip file and irrelevant contents
+# unlink(zip_dest_22)
```
Uncommented

vpancini commented:
In the 2022 project update, we used data from County Health Rankings for this metric. They suppressed values for counties with 0 reported primary care physicians but population greater than 2,000. I could not find an explanation of this in their documentation, but I think it is because they're assuming that counties with sizable populations but 0 reported PCPs are errors, whereas counties with populations less than 2,000 and 0 PCPs are true 0s. This made it so that, in the last update, counties with 0 PCPs had a quality value of either `1` or `NA` (counties with population greater than 2,000 and 0 PCPs had a ratio of `0` and a quality of `NA`; Counties with population less than 2,000 and 0 PCPs had a ratio of `0` and a quality of `1`). In this update, we do not apply this suppression rule, so all counties with 0 reported PCPs are assigned a quality indicator of `NA` even though some may be true zeros. Future updates may want to consider ways to differentiate these cases. This is also explained in the `.qmd` file.

malcalakovalski commented on 04_health/access-health-services/access-health-services-all.qmd:
````diff
@@ -0,0 +1,829 @@
+
+```{r load-chr}
+# Download file directly from internet
+download.file(
+  url = "https://www.countyhealthrankings.org/sites/default/files/media/document/2023%20County%20Health%20Rankings%20Data%20-%20v2.xlsx",
+  destfile = "04_health/access-health-services/data/chr2023.xlsx",
+  mode = "wb"
+)
````
This didn't run for me because it's missing `here::here` in `destfile`. This fixes it:

```suggestion
 download.file(
  url = "https://www.countyhealthrankings.org/sites/default/files/media/document/2023%20County%20Health%20Rankings%20Data%20-%20v2.xlsx",
  destfile = here::here("04_health", "access-health-services", "data", "chr2023.xlsx"),
  mode = "wb"
)
```

vpancini replied:
Fixed!

malcalakovalski commented on 04_health/access-health-services/access-health-services-all.qmd:
````diff
@@ -0,0 +1,829 @@
+We download the Vintage 2020 population estimates and are able to recreate their ratio.
+```{r load-vintage-2020-pop}
+# Download the Vintage 2020 population estimates 
+download.file(
+  url = "https://www2.census.gov/programs-surveys/popest/datasets/2010-2020/counties/totals/co-est2020.csv",
+  destfile = "04_health/access-health-services/data/chr_pop.csv",
+  mode = "wb"
````
Same thing as before here:

``` suggest
download.file(
  url = "https://www2.census.gov/programs-surveys/popest/datasets/2010-2020/counties/totals/co-est2020.csv",
  destfile = here::here("04_health", "access-health-services", "data", "chr_pop.csv"),
  mode = "wb"
)
```

vpancini replied:
Fixed!

malcalakovalski reviewed (changes requested):
Everything looks great now, thanks for addressing all my suggestions! A few more things and this will be good to go:

- I commented on two chunks you need to edit to make the `.qmd` reproducible
- It would be best to remove old files like `access-health-services-all.Rmd`. 
- A `.gitignore` file that ignores the data folder should be added to the metric folder (`04_health/access-health-services`). I see you commited changes to the project gitignore, but I think this would also be useful . You could also ignore some of the files generated by rendering the `.qmd`.

vpancini commented:
> Everything looks great now, thanks for addressing all my suggestions! A few more things and this will be good to go:
> 
> * I commented on two chunks you need to edit to make the `.qmd` reproducible
> * It would be best to remove old files like `access-health-services-all.Rmd`.
> * A `.gitignore` file that ignores the data folder should be added to the metric folder (`04_health/access-health-services`). I see you commited changes to the project gitignore, but I think this would also be useful . You could also ignore some of the files generated by rendering the `.qmd`.

- Implemented these edits, and also added code to only download those files if they have not been already downloaded.
- Good call. Removed old files.
- Good call. Added a new `.gitignore` file to the metric folder.

malcalakovalski reviewed (approved):
Everything looks good and all my comments have been addressed!
